### PR TITLE
Hancheng default commons UI

### LINKED ISSUE #7: FEATURE: Refactor CommonList component so that it has a default UI
### Have you updated the title?

- [X] I have updated the issue's title.

### As a...

user

### I can...

easily tell when there are no commons for me to visit or join

### So that...

I won't get confused when common list component is empty

### Discussion

1. Currently, when common list is empty, the UI only shows the names of the columns and a white background.
2. Improve UX for ```CommonList.js``` by adding a default UI to say something like "currently there are no commons available" so that users can better understand what is going on.
3. The design of this feature is entirely up to students.

### Acceptance Criteria

- [ ] When the common list component in home page is empty, we should show a default UI.
- [ ] When the common list component is not empty, we should the list component as it is.

### Implementation Todos

Frontend:
- [ ] Design a default for common list when the list is empty.

Testing:
- [ ] When the common list component in home page is empty, we should show a default UI.
- [ ] When the common list component is not empty, we should the list component as it is.

### PR DESCRIPTION
**Pull request by hancheng-li**
## Overview

In this PR, I added a message saying "Currently, there are no commons available" when there are no commons to join or visit. 

## Screenshots
Screenshot 2023-11-21 at 1 29 29 PM
Screenshot 2023-11-21 at 1 29 04 PM
Screenshot 2023-11-21 at 1 28 24 PM

## Tests

- [x] Backend Unit tests (`mvn test`) pass
- [x] Backend Test coverage (`mvn test jacoco:report`) 100%
- [x] Backend Mutation tests (`mvn test pitest:mutationCoverage`) 100% 
- [x] Frontend Unit tests (`npm test`) pass
- [x] Frontend Test coverage (`npm run coverage`) 100%
- [x] Frontend Mutation tests (`npx stryker run`) 100% 
- [x] Frontend Linting (`npx eslint --fix src`) 

## Linked Issues

Closes #7

### CODE REVIEW

File: frontend/src/main/components/OurTable.js [hunk 1]
```diff
@@ -57,9 +57,12 @@ export default function OurTable({ columns, data, testid = "testid", ...rest })
           return (
             <tr {...row.getRowProps()}>
               {row.cells.map((cell, _index) => {
+                const cellStyle = { textAlign: 'right' };
```
Comment (pconrad) [inline frontend/src/main/components/OurTable.js:60]: This should not be done at the OurTable level, and doesn't seem like it's part of this PR in any case.

You may want to revert this change.

File: frontend/src/main/components/Reports/ReportHeaderTable.js [hunk 2]
```diff
@@ -6,14 +6,17 @@ export default function ReportHeaderTable({ report  }) {
         {
             Header: 'Cow Price',
             accessor: 'cowPrice', 
+            Cell: ({ value }) => `$${value}`
```
Comment (pconrad) [inline frontend/src/main/components/Reports/ReportHeaderTable.js:9]: This also feels like it's unrelated to this PR.

You may want to revert this change.

It might actually be easier to just make a new branch with only the changes that are relevant to the specific issue at hand.

File: frontend/src/main/components/Reports/ReportLineTable.js [hunk 3]
```diff
@@ -14,6 +14,7 @@ export default function ReportLineTable({ reportLines }) {
         {
             Header: 'Total Wealth',
             accessor: 'totalWealth',
+            Cell: ({ value }) => `$${value}`
```
Comment (pconrad) [inline frontend/src/main/components/Reports/ReportLineTable.js:17]: Not sure why changes to this file are in this PR.  I feel like this is going to become a theme...

File: frontend/src/tests/components/Commons/CommonsList.test.js [hunk 4]
```diff
@@ -89,4 +89,8 @@ describe("CommonsList tests", () => {
             i++;
         })
     });
+    test('displays default message when there are no commons', () => {
```
Comment (pconrad) [inline frontend/src/tests/components/Commons/CommonsList.test.js:92]: This is correct!   I wonder if the only changes that should be in this PR are changes to this file and to `CommonsList.js`?

File: src/main/java/edu/ucsb/cs156/happiercows/entities/ReportLine.java [hunk 5]
```diff
@@ -7,7 +7,9 @@
 import org.hibernate.annotations.CreationTimestamp;
 
 import lombok.Data;
+import lombok.Getter;
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/happiercows/entities/ReportLine.java:10]: This definitely seems unrelated to the PR at hand.. wondering why this is here?

**Review by pconrad — CHANGES_REQUESTED** [review]
While this doesn't yet have a peer code review, @GlassBricks brought this to my attention...

My sense is that this looks good if we only focus on the changes to `CommonsList.js` and `CommonsList.test.js`.  That's probably all that should be in this PR.

When making a PR, we typically want to start from a clean and updated version of the `main` branch *except* in the case where we are depending on changes to an earlier PR.  I don't think that's the case here.  If I'm mistaken, please let me know.

Otherwise, consider cleaning up this branch. It may be easier to start with a clean version of main on a new branch and just apply the changes to those two files, using this as a reference, and then closing  this PR.

**Comment by GlassBricks** [comment]
As a note, you can include the other changes (adding dollar sign, table right-alignment) in separate PR(s).
To do that, it might be easiest to start a new branch(es), then copy over the changes.

**Comment by hancheng-li** [comment]
> While this doesn't yet have a peer code review, @GlassBricks brought this to my attention...
> 
> My sense is that this looks good if we only focus on the changes to `CommonsList.js` and `CommonsList.test.js`. That's probably all that should be in this PR.
> 
> When making a PR, we typically want to start from a clean and updated version of the `main` branch _except_ in the case where we are depending on changes to an earlier PR. I don't think that's the case here. If I'm mistaken, please let me know.
> 
> Otherwise, consider cleaning up this branch. It may be easier to start with a clean version of main on a new branch and just apply the changes to those two files, using this as a reference, and then closing this PR.

I think those modified files are from my previous PR which hasn't been merged yet. That PR wants to add the $ sign in front of quantities with currency involved and right aligns all the cells.

**Comment by GlassBricks** [comment]
> I think those modified files are from my previous PR which hasn't been merged yet. That PR wants to add the $ sign in front of quantities with currency involved and right aligns all the cells.

It's best for PRs to be independent, you should start new branches from `origin/main`.
However, here's how you can clean this up from here:

With the `Hancheng-DefaultCommonsUI` branch checked out, run
```
git rebase --onto origin/main Hangcheng-InstructorReport
git push --force
```
The rebase takes the commits from the current branch up until Hangcheng-InstructorReport, then "reattaches" them to the main branch.

**Comment by hancheng-li** [comment]
> > I think those modified files are from my previous PR which hasn't been merged yet. That PR wants to add the $ sign in front of quantities with currency involved and right aligns all the cells.
> 
> It's best for PRs to be independent, you should start new branches from `origin/main`. However, here's how you can clean this up from here:
> 
> With the `Hancheng-DefaultCommonsUI` branch checked out, run
> 
> ```
> git rebase --onto origin/main Hangcheng-InstructorReport
> git push --force
> ```
> 
> The rebase takes the commits from the current branch up until Hangcheng-InstructorReport, then "reattaches" them to the main branch. The push with --force allows changes that aren't "forward-only" changes, this is often needed after a rebase.

Thanks for the information. I've just rebased my branch onto my previous one and will do the same thing in the future.

File: frontend/src/main/components/Leaderboard/LeaderboardTable.js [hunk 6]
```diff
@@ -24,10 +24,16 @@ export default function LeaderboardTable({ leaderboardUsers , currentUser }) {
             accessor: (row, _rowIndex) => {
                 return USD.format(row.totalWealth);
             },
+            // sortType:
```
Comment (pconrad) [inline frontend/src/main/components/Leaderboard/LeaderboardTable.js:27]: Please see: https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

**Review by pconrad — CHANGES_REQUESTED** [review]
There is commented out code.  Please see:

https://ucsb-cs156.github.io/topics/code_reviews/commented_out_code.html

File: frontend/src/main/components/Leaderboard/LeaderboardTable.js (hunk 6 above)
Comment (hancheng-li) [reply frontend/src/main/components/Leaderboard/LeaderboardTable.js]: Got it, I will delete the commented-out code for this PR.

**Review by pconrad — APPROVED** [review]
LGTM
